pkg/imgpkg/cmd/copy.go

```diff
@@ -242,17 +243,42 @@ func (o *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
 func processedImagesMediaType(processedImages *ctlimgset.ProcessedImages) []string {
 	everyMediaType := []string{}
 	for _, image := range processedImages.All() {
-		layers, err := image.Image.Layers()
-		if err != nil {
-			continue
-		}
-		for _, layer := range layers {
-			mediaType, err := layer.MediaType()
+		if image.ImageIndex != nil {
+			indexManifest, err := image.ImageIndex.IndexManifest()
 			if err != nil {
 				continue
 			}
-			everyMediaType = append(everyMediaType, string(mediaType))
+			for _, descriptor := range indexManifest.Manifests {
+				image, err := image.ImageIndex.Image(descriptor.Digest)
```

Comment:
i might be wrong, i thought that indexes could contain other indexes as well.

Reply:
Added logic to recurse through image indexes